src/adapter/src/catalog.rs

```diff
@@ -101,6 +101,18 @@ pub static SYSTEM_USER: Lazy<User> = Lazy::new(|| User {
     external_metadata: None,
 });
 
+pub static INTROSPECTION_USER: Lazy<User> = Lazy::new(|| User {
+    name: "mz_introspection".into(),
+    external_metadata: None,
+});
+
+pub static INTERNAL_USER_NAMES: Lazy<BTreeSet<String>> = Lazy::new(|| {
```

Comment:
This seems a bit pointless. A BTree of size 2 is just going to be the same as a vector. Let me know if you think I should change it to a vec or something else.